gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/desktop/application/tasks/AbstractCheckNativeDistributionRuntime.kt

```diff
@@ -69,18 +78,37 @@ abstract class AbstractCheckNativeDistributionRuntime : AbstractComposeDesktopTa
         val jpackageExecutabke = jdkHome.getJdkTool("jpackage")
         ensureToolsExist(javaExecutable, jlinkExecutable, jpackageExecutabke)
 
-        val jvmRuntimeVersionString = getJavaRuntimeVersion(javaExecutable)
+        val jdkRuntimeProperties = getJDKRuntimeProperties(javaExecutable)
 
-        val jvmRuntimeVersion = jvmRuntimeVersionString?.toIntOrNull()
-            ?: jdkDistributionProbingError("JDK version '$jvmRuntimeVersionString' has unexpected format")
+        val jdkMajorVersionString = jdkRuntimeProperties.getProperty(JdkVersionProbe.JDK_MAJOR_VERSION_KEY)
+        val jdkMajorVersion = jdkMajorVersionString?.toIntOrNull()
+            ?: jdkDistributionProbingError("JDK version '$jdkMajorVersionString' has unexpected format")
```

Comment:
Will JDK "1.8" parses correctly?
Or maybe because we don't support 1.8 it is no matter...

Reply:
The probe returns only major version, which is always an integer!